k8s/sonarqube/README.md

```diff
@@ -0,0 +1,442 @@
+
+You can remove existing metric descriptors using
+[Stackdriver's REST API](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors/delete).
+
+# Scaling
+
+SonarQube Community Edition doest not support scaling.
+
+# Backup and restore
+
+[From official documentation](https://docs.sonarqube.org/7.6/architecture/architecture-integration/)
+The SonarQube Platform is made of 4 components:
+
+- One SonarQube Server starting 3 main processes:
+
+   - Web Server for developers, managers to browse quality snapshots and configure the SonarQube instance
+   - Search Server based on Elasticsearch to back searches from the UI
+   - Compute Engine Server in charge of processing code analysis reports and saving them in the SonarQube Database
+
+- One SonarQube Database to store:
+
+   - the configuration of the SonarQube instance (security, plugins settings, etc.)
+   - the quality snapshots of projects, views, etc.
+
+- Multiple SonarQube Plugins installed on the server, possibly including language, SCM, integration, authentication, and governance plugins
+
+- One or more SonarScanners running on your Build / Continuous Integration Servers to analyze projects
+
+For our application database the most important place, `plugins` folder stored on PVC (Persistent Volume Claim).
+
+## Backing up plugin and data
+
+This shell script will create copy of plugins folder in current folder
+```shell
+kubectl --namespace $NAMESPACE cp $(kubectl -n$NAMESPACE get pod -oname | \
+                sed -n /\\/$APP_INSTANCE_NAME-sonarqube/s.pods\\?/..p):/opt/sonarqube/extensions/ ./
+```
+
+This shell script will create copy of content from folder `data` in current folder:
+
+```shell
+kubectl --namespace $NAMESPACE cp $(kubectl -n$NAMESPACE get pod -oname | \
+                sed -n /\\/$APP_INSTANCE_NAME-sonarqube/s.pods\\?/..p):/opt/sonarqube/data ./
+```
+
+## Backing up PostgreSQL
+
+This shell script will create `postgresql-backup.sql` dump of all DB in PostgreSQL.
+
+```shell
+kubectl --namespace $NAMESPACE exec -t \
+	$(kubectl -n$NAMESPACE get pod -oname | \
+		sed -n /\\/$APP_INSTANCE_NAME-postgresql-deployment/s.pods\\?/..p) \
+	-c postgresql-server \
+	-- pg_dumpall -c -U postgres > backup.sql
+```
+
+## Restoring your PostgreSQL
+
+This shell script will prepare dump for restore
+
+```shell
+sed -i '1 i\REVOKE CONNECT ON DATABASE sonar FROM PUBLIC;' backup.sql
+sed -i '1 a select pg_terminate_backend(pid) from pg_stat_activity where datname='sonar';' backup.sql
+echo "GRANT CONNECT ON DATABASE sonar TO PUBLIC;" >> backup.sql
+```
+
+This shell script will restore dump `backup.sql` to PostgreSQL
+
+```shell
+cat backup.sql | kubectl --namespace $NAMESPACE exec -i \
+	$(kubectl -n$NAMESPACE get pod -oname | \
+		sed -n /\\/$APP_INSTANCE_NAME-postgresql-deployment/s.pods\\?/..p) \
+	-c postgresql-server \
+	-- psql -U postgres
+```
+
+This shell script will copy files from current folder to folder `data` in pod
```

Comment:
I do not like `pod sonarqube`. `sonarqube` is not a name of that pod, it's a pod running SonarQube application.

Reply:
part "Restoring" was rewritten